nixos/modules/services/databases/postgresql.nix

```
@@ -155,7 +156,50 @@ in
       };
 
       ensureDatabases = mkOption {
-        type = types.listOf types.str;
+        type = types.listOf (types.oneOf [types.str (types.submodule {
+          options = {
+            name = mkOption {
+              type = types.str;
+              description = ''
+                Name of the database to ensure.
+              '';
+            };
+            withOptions = mkOption {
+              type = types.attrsOf types.str;
+              description = ''
+                The options to pass to the `CREATE DATABASE` statement.
+                The keys are the options and the values are the values to pass.
+
+                See: https://www.postgresql.org/docs/current/sql-createdatabase.html
+
+                [ WITH ] [ OWNER [=] user_name ]
+                      [ TEMPLATE [=] template ]
+                      [ ENCODING [=] encoding ]
+                      [ STRATEGY [=] strategy ]
+                      [ LOCALE [=] locale ]
+                      [ LC_COLLATE [=] lc_collate ]
+                      [ LC_CTYPE [=] lc_ctype ]
+                      [ ICU_LOCALE [=] icu_locale ]
+                      [ ICU_RULES [=] icu_rules ]
+                      [ LOCALE_PROVIDER [=] locale_provider ]
+                      [ COLLATION_VERSION = collation_version ]
+                      [ TABLESPACE [=] tablespace_name ]
+                      [ ALLOW_CONNECTIONS [=] allowconn ]
+                      [ CONNECTION LIMIT [=] connlimit ]
+                      [ IS_TEMPLATE [=] istemplate ]
+                      [ OID [=] oid ]
```

Just link to the upstream documentation rather than duplicate part of it into NixOS.

```suggestion
                See [PostreSQL's documentation](https://www.postgresql.org/docs/current/sql-createdatabase.html).
```

---

I just want someone who trying to search these keywords can find this option.    But maybe you are right, I'll remove them.